brokerapi/brokers/cloudsql/broker.go

```diff
@@ -266,23 +281,32 @@ func (b *CloudSQLBroker) Provision(instanceId string, details models.ProvisionDe
 // executing this "synchronously" even though technically db creation returns an op - but it's just a db call, so
 // it should be quick and not actually async.
 func (b *CloudSQLBroker) FinishProvisioning(instanceId string, params map[string]string) error {
-
 	var err error
+
+	serviceInstanceDetails := models.ServiceInstanceDetails{}
+	if err = db_service.DbConnection.Where("ID = ?", instanceId).First(&serviceInstanceDetails).Error; err != nil {
+		return models.ErrInstanceDoesNotExist
+	}
+
 	sqlService, err := googlecloudsql.New(b.Client)
 	if err != nil {
 		return fmt.Errorf("Error creating new CloudSQL Client: %s", err)
 	}
 
 	dbService := googlecloudsql.NewInstancesService(sqlService)
-	clouddb, err := dbService.Get(b.ProjectId, params["instance_name"]).Do()
+	clouddb, err := dbService.Get(b.ProjectId, serviceInstanceDetails.Name).Do()
 	if err != nil {
 		return fmt.Errorf("Error getting instance from api: %s", err)
 	}
 
 	//create actual database entry
+	var cloudSqlOperation CloudSqlOperation
+	if err := json.Unmarshal([]byte(serviceInstanceDetails.OtherDetails), &cloudSqlOperation); err != nil {
+		return fmt.Errorf("Error unmarshalling operation status details: %s", err)
+	}
 
 	d := googlecloudsql.Database{
-		Name: params["database_name"],
+		Name: cloudSqlOperation.DatabaseName,
```

Comment:
I think a cleaner way to do this would be to generate the database name in this method instead of saving it on the CloudSQLOperation object.